### PR TITLE
fix: config jest for unit testing

### PR DESCRIPTION
**Pull request by CodesOfRa**
#### Please provide a brief summary of this pull request.
Set up Jest config for unit testing
#### If this is a new feature, have you updated the documentation?
Done

### CODE REVIEW

File: ui/package.json [hunk 1]
```diff
@@ -5,27 +5,30 @@
   "scripts": {
     "serve": "vue-cli-service serve",
     "build": "vue-cli-service build --mode production --target lib --report --name FundamentalVue src/main.ts",
+    "test": "vue-cli-service test:unit",
     "lint": "vue-cli-service lint",
```
Comment (bcullman) [inline ui/package.json]: Same as test. Either remove, or point to test.

File: ui/package.json [hunk 2]
```diff
@@ -5,27 +5,30 @@
...
-    "test": "vue-cli-service test:unit \"src/**/__tests__/*.test.{ts,js,vue}\"",
+    "test:unit": "vue-cli-service test:unit",
```
Comment (bcullman) [inline ui/package.json]: Same as test. Either remove, or point to test.

Comment (CodesOfRa) [reply ui/package.json]: removed :)

**Comment by bcullman** [comment]
not sure how yarn works, but generally, when there is an change in package.json, there is a corresponding change in package-lock.json, but not seeing it as part of the PR.

**Comment by CodesOfRa** [comment]
@bcullman Not very sure how yarn works either :) . I thought `yarn.lock` is enough :) . I'll ask @ChristianKienle

**Comment by greg-a-smith** [comment]
@bcullman @CodesOfRa My understanding is that `package-lock.json` file is an "NPM" thing and `yarn.lock` is a "Yarn" thing.  I think Yarn supports the reading of `package-lock.json`, but ultimately writes the `yarn.lock` file when there are changes.  I see this repo has both lock files and it probably shouldn't (or both should be kept up-to-date, but that sounds like a lot of extra work).  I think that most devs on this repo have been using yarn so maybe the `package-lock.json` file should be removed?

**Comment by ChristianKienle** [comment]
> @bcullman @CodesOfRa My understanding is that `package-lock.json` file is an "NPM" thing and `yarn.lock` is a "Yarn" thing. I think Yarn supports the reading of `package-lock.json`, but ultimately writes the `yarn.lock` file when there are changes in `package.json`. I see this repo has both lock files and it probably shouldn't (or both should be kept up-to-date, but that sounds like a lot of extra work). I think that most devs on this repo have been using yarn so maybe the `package-lock.json` file should be removed?

Yes. package-lock files should be removed. Yarn will even complain about package-lock-files if it sees any.

**Comment by ChristianKienle** [comment]
@CodesOfRa I have pushed directly to this branch. I hope you don't mind. If you dislike my changes I will revert them.

- I have deleted the package-lock.json-file
- I have changed the dependencies slightly. I did this so that we have less warnings and that they are more similar to what vue-cli generates currently.
- I have added two args to the test script:
  - `--runInBand`: Everyone says using this is bad but I suggest we turn this on until executing the tests becomes too slow. This effectively causes tests to be run serially and not concurrently. This makes the execution of our tests more predictive. Without this option some tests may run concurrently and thus seeing changes in performance is very hard.
  - `--verbose`: I like the verbose output much more. If you don't I will of course revert :D

**Pull request merged**